homeassistant/components/androidtv_remote/media_player.py

```diff
@@ -0,0 +1,185 @@
+"""Media player support for Android TV Remote."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from androidtvremote2 import AndroidTVRemote
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import AndroidTVRemoteBaseEntity
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Android TV media player entity based on a config entry."""
+    api: AndroidTVRemote = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AndroidTVRemoteMediaPlayerEntity(api, config_entry)])
+
+
+class AndroidTVRemoteMediaPlayerEntity(AndroidTVRemoteBaseEntity, MediaPlayerEntity):
+    """Android TV Remote Media Player Entity."""
+
+    _attr_assumed_state = True
+    _attr_device_class = MediaPlayerDeviceClass.TV
+    _attr_supported_features = (
+        MediaPlayerEntityFeature.PAUSE
+        | MediaPlayerEntityFeature.VOLUME_STEP
+        | MediaPlayerEntityFeature.VOLUME_MUTE
+        | MediaPlayerEntityFeature.VOLUME_SET
+        | MediaPlayerEntityFeature.PREVIOUS_TRACK
+        | MediaPlayerEntityFeature.NEXT_TRACK
+        | MediaPlayerEntityFeature.TURN_ON
+        | MediaPlayerEntityFeature.TURN_OFF
+        | MediaPlayerEntityFeature.PLAY
+        | MediaPlayerEntityFeature.STOP
+        | MediaPlayerEntityFeature.PLAY_MEDIA
+    )
+
+    def __init__(self, api: AndroidTVRemote, config_entry: ConfigEntry) -> None:
+        """Initialize the entity."""
+        super().__init__(api, config_entry)
+
+        self._volume_max: int | None = None
+        self._volume_set_task: asyncio.Task | None = None
+        self._channel_set_task: asyncio.Task | None = None
+
+        self._update_current_app(api.current_app)
+        self._update_volume_info(api.volume_info)
+
+        @callback
+        def current_app_updated(current_app: str) -> None:
+            self._update_current_app(current_app)
+            self.async_write_ha_state()
+
+        @callback
+        def volume_info_updated(volume_info: dict[str, str | bool]) -> None:
+            self._update_volume_info(volume_info)
+            if not self._volume_set_task or self._volume_set_task.done():
+                self.async_write_ha_state()
+
+        api.add_current_app_updated_callback(current_app_updated)
+        api.add_volume_info_updated_callback(volume_info_updated)
+
+    def _update_current_app(self, current_app: str) -> None:
+        """Update current app info."""
+        self._attr_app_id = current_app
+        self._attr_app_name = current_app
```

Comment:
Skip setting the app name for now? How is it used?

Reply:
I keep `app_name` to show the app name in the frontend card (see screen above).